lisk-migrator is UNABLE to save `node Forging Status information` to the disk

### Expected behavior

lisk-migrator can save `node Forging Status information` to the disk

### Actual behavior

```bash
Unable to save the node Forging Status information to the disk, please find it below instead:
[
  {
    "address": "da0fe279669a8395d58e47a7fcda5bfbdb37f82b",
    "forging": false
  },
  {
    "address": "dea5007ff22309f3687da052b5bd8b7ae29e1ab6",
    "forging": false
  },
  {
    "address": "62dbc020b55bbb629b11f6cdb585c4f97028666e",
    "forging": false
  }
]
```

### Steps to reproduce

I used a custom height to test the tool.

CMD: `lisk-migrator --snapshot-height 20436848 --output ~/.lisk/lisk-core/config/testnet --auto-migrate-config --auto-start-lisk-core-v4`

### Which version(s) does this affect? (Environment, OS, etc...)

OS: Ubuntu 20

// node -v
v18.18.0
// npm -v
9.8.1
// lisk-core -v
lisk-core/3.1.0-rc.1.337536d linux-x64 node-v18.16.1

### FULL TERMINAL LOG

```bash
USERNAME@HOSTNAME:~$ lisk-migrator --snapshot-height 20436848 --output ~/.lisk/lisk-core/config/testnet --auto-migrate-config --auto-start-lisk-core-v4
Verifying if backup height from node config matches snapshot height... Snapshot height matches backup height
Verifying snapshot height to be multiples of round length i.e 103... Snapshot height is valid
Verifying Lisk Core version... 3.1.0-rc.1.337536d detected
Compiling Lisk Core configuration... done
Loading Lisk Core configuration... done
Waiting for snapshot height to be finalized: [============------------------] 40% | Remaining: 153/259 | Height: 20436695/20436848 | ETA: 25m 30s
Unable to save the node Forging Status information to the disk, please find it below instead:
[
  {
    "address": "da0fe279669a8395d58e47a7fcda5bfbdb37f82b",
    "forging": false
  },
  {
    "address": "dea5007ff22309f3687da052b5bd8b7ae29e1ab6",
    "forging": false
  },
  {
    "address": "62dbc020b55bbb629b11f6cdb585c4f97028666e",
    "forging": false
  }
]
Waiting for snapshot height to be finalized: [==============================] 100% | Remaining: -2/259 | Height: 20436850/20436848 | ETA: 0
Creating database instance... done
Creating genesis assets... done
Exporting genesis block to the path /home/USERNAME/.lisk/lisk-core/config/testnet/15f0dacc1060e91818224a94286b13aa04279c640bd5d6f193182031d133df7c... done
Creating backup for old config... done
Setting config property 'system.version' to: 0.1.0.... done
Setting config property 'system.logLevel' to: info.... done
Setting config property 'network.port' to: 7001.... done
Calculating and updating config property 'genesis.minimumCertifyHeight'.... done
Validating migrated user configuration... done
Exporting user configuration to the path: /home/USERNAME/.lisk/lisk-core/config/testnet/15f0dacc1060e91818224a94286b13aa04279c640bd5d6f193182031d133df7c... done
Installing lisk-core v4... !
Error: Command failed: npm i -g lisk-core@^4.0.0-rc.1
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/lib/node_modules/lisk-core
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/lib/node_modules/lisk-core'
npm ERR!  [Error: EACCES: permission denied, mkdir '/usr/lib/node_modules/lisk-core'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/lib/node_modules/lisk-core'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.
npm ERR! A complete log of this run can be found in: /home/USERNAME/.npm/_logs/2023-09-25T00_31_46_670Z-debug-0.log
    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)
```

--- a/src/events.ts
+++ b/src/events.ts
@@ -33,7 +33,7 @@ export const captureForgingStatusAtSnapshotHeight = (
 			if (forgingStatus.length) {
 				const forgingStatusJsonFilepath = resolve(outputDir, 'forgingStatus.json');
 				try {
-					await write(forgingStatusJsonFilepath, JSON.stringify(forgingStatus, null, 2));
+					await write(forgingStatusJsonFilepath, JSON.stringify(forgingStatus, null, '\t'));
 					_this.log(`\nFinished exporting forging status to ${forgingStatusJsonFilepath}.`);
 				} catch (error) {
 					_this.log(

--- a/src/index.ts
+++ b/src/index.ts
@@ -138,7 +138,10 @@ class LiskMigrator extends Command {
 			cli.action.stop('Snapshot height is valid');
 
 			const networkConstant = NETWORK_CONSTANT[networkIdentifier] as NetworkConfigLocal;
-			const outputDir = `${outputPath}/${networkIdentifier}`;
+			const outputDir = flags.output ? outputPath : `${outputPath}/${networkIdentifier}`;
+
+			// Ensure the output directory is present
+			if (fs.existsSync(outputDir)) fs.mkdirSync(outputDir, { recursive: true });
 
 			// Asynchronously capture the node's Forging Status information at the snapshot height
 			// This information is necessary for the node operators to enable generator post-migration without getting PoM'd

--- a/src/index.ts
+++ b/src/index.ts
@@ -63,7 +63,7 @@ class LiskMigrator extends Command {
 			char: 'o',
 			required: false,
 			description:
-				'File path to write the genesis block json. If not provided, it will default to cwd/genesis_block.json.',
+				'File path to write the genesis block. If not provided, it will default to cwd/output/{v3_networkIdentifier}/genesis_block.blob.',
 		}),
 		'lisk-core-v3-data-path': flagsParser.string({
 			char: 'd',

--- a/src/index.ts
+++ b/src/index.ts
@@ -306,7 +306,7 @@ class LiskMigrator extends Command {
 				}
 			} else {
 				this.log(
-					`Please copy ${snapshotDirPath} directory to the Lisk Core V4 data directory in order to access legacy blockchain information.`,
+					`Please copy the contents of ${snapshotDirPath} directory to 'data/legacy.db' under the Lisk Core V4 data directory (e.g: ~/.lisk/lisk-core/data/legacy.db/) in order to access legacy blockchain information.`,
 				);
 				this.log('Please copy genesis block to the Lisk Core V4 network directory.');
 			}

--- a/src/utils/genesis_block.ts
+++ b/src/utils/genesis_block.ts
@@ -64,9 +64,6 @@ export const writeGenesisAssets = async (
 	genesisAssets: GenesisAssetEntry[],
 	outputDir: string,
 ): Promise<void> => {
-	if (fs.existsSync(outputDir)) fs.rmdirSync(outputDir, { recursive: true });
-	fs.mkdirSync(outputDir, { recursive: true });
-
 	const genesisAssetsJsonFilepath = path.resolve(outputDir, 'genesis_assets.json');
 	fs.writeFileSync(
 		genesisAssetsJsonFilepath,

